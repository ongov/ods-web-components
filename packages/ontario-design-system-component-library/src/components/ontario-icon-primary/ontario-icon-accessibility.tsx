import { Component, Prop, h } from '@stencil/core';
import accessibility from './assets/ontario-icon-accessibility.svg'

/** @internal **/
@Component({
  tag: 'ontario-icon-accessibility',
  styleUrl: 'ontario-icon-primary.scss',
  shadow: false,
})

export class OntarioIconAccessibility {

  @Prop() colour: "black" | "blue" | "grey";

  render() {
    return <div class={`ontario-icon ontario-icon--${this.colour}`} innerHTML={accessibility} />;
  }
};
