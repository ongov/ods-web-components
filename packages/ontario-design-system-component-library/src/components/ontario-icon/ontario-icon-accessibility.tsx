// content automatically generated by `generate-icons.js` begins
import { Component, Prop, h, Watch, State } from '@stencil/core';
import { IconWithColour } from './icon.interface';
import { IconSize, IconColour } from './ontario-icon.enum';
import { ConsoleType, MessageStyle } from '../../utils/console-message/console-message.enum';
import { printConsoleMessage } from '../../utils/console-message/console-message';

@Component({
    tag: 'ontario-icon-accessibility',
    styleUrl: 'ontario-icon.scss',
    shadow: true,
})
export class OntarioIconAccessibility implements IconWithColour {
    /**
     * The icon width will autogenerate the height since the icons are in square format, thus preserving
     * the aspect ratio.
     */
    @Prop() iconWidth: number = IconSize.DEFAULT;

    /**
	 * Mutable variable, for internal use only.
	 * Set the icon's width depending on validation result.
	 */
	@State() iconWidthState: number = IconSize.DEFAULT;

    /**
     * Watch for changes in the `iconWidth` variable for validation purpose.
     * If the user input is not a number or is a negative number then `iconWidth` will be set to its default (24).
     */
    @Watch('iconWidth')
    validateWidth() {
		if (isNaN(this.iconWidth) || (!isNaN(this.iconWidth) && this.iconWidth <= 0)) {
			printConsoleMessage([
        {
            message: '%c icon-width ',
            style: MessageStyle.CODE,
        },
        {
            message: '%con',
            style: MessageStyle.REGULAR,
        },
        {
            message: `%c <ontario-icon-accessibility> `,
            style: MessageStyle.CODE,
        },
        {
            message: `%cwas set to a non-numeric value; only positive number is allowed. The default size of`,
            style: MessageStyle.REGULAR,
        },
        {
            message: '%c 24px ',
            style: MessageStyle.CODE,
        },
        {
            message: '%cwas assumed.',
            style: MessageStyle.REGULAR,
        },
    ], ConsoleType.WARNING);
			this.iconWidthState = IconSize.DEFAULT;
		} else {
			this.iconWidthState = this.iconWidth;
		}
    }
    
    /**
     * Set the icon's colour.
     */
    @Prop() colour: IconColour = IconColour.BLACK;

    /**
	 * Mutable variable, for internal use only.
	 * Set the icon's colour based on validation result.
	 */
	@State() iconColourState: IconColour = IconColour.BLACK;

    /**
     * Watch for changes in the `colour` variable for validation purpose.
     * If the user input doesn't match one of the enum values then `colour` will be set to its default (`black`).
     * If a match is found in one of the enum values then `colour` will be set to the matching enum value.
     */
    @Watch('colour')
    validateColour() {
        this.iconColourState = (this.colour && Object.values(IconColour).find(colour => colour === this.colour.toLowerCase())) || IconColour.BLACK;
    }
    
	/**
	 * Stencil component lifecycle method that is called once after the component is first connected to the DOM.
	 */
    componentWillLoad() {
        this.validateColour();
        this.validateWidth();
    }

    /**
	 * Returns the HTML code to be rendered into a custom element.
	 */
    render() {
        return (
            <div class={`ontario-icon ontario-icon--${this.iconColourState}`} style={{ 'width': `${this.iconWidthState}px` }}>
                <svg class="svg-icon" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" id="accessibility"><path d="M18.4 11.2l-4.1.2 2.3-2.6c.2-.3.3-.8.2-1.3-.1-.3-.2-.6-.5-.8l-5.4-3.2c-.4-.3-1-.2-1.4.1L6.8 6.1c-.5.5-.6 1.2-.1 1.7.4.5 1.2.5 1.7.1l2-1.8 1.9 1.1-4.2 4.3c-.1.1-.1.2-.2.2-.5.2-1 .4-1.4.7L8 13.9c.5-.2 1-.4 1.5-.4 1.9 0 3.5 1.6 3.5 3.5 0 .6-.1 1.1-.4 1.5l1.5 1.5a5.29 5.29 0 0 0 .9-3c0-1.2-.4-2.4-1.1-3.3l3.3-.3-.2 4.8c-.1.7.4 1.2 1.1 1.3h.1c.6 0 1.1-.5 1.2-1.1l.2-5.9c0-.3-.1-.7-.3-.9-.3-.3-.6-.4-.9-.4zM18 5.5a2 2 0 0 0 2-2 2 2 0 0 0-2-2 2 2 0 0 0-2 2 2 2 0 0 0 2 2zm-5.5 16.1a5.29 5.29 0 0 1-3 .9C6.5 22.5 4 20 4 17a5.29 5.29 0 0 1 .9-3l1.5 1.5c-.2.5-.4 1-.4 1.5 0 1.9 1.6 3.5 3.5 3.5.6 0 1.1-.1 1.5-.4l1.5 1.5z"/></svg>
            </div>
        );
    }
};
// content automatically generated by `generate-icons.js` ends
