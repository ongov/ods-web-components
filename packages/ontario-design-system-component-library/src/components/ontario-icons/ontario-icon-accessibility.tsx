import { Component, Prop, h } from '@stencil/core';
import accessibility from './assets/ontario-icon-accessibility.svg'

/** @internal **/
@Component({
  tag: 'ontario-icon-accessibility',
  styleUrl: 'ontario-icons.scss',
  shadow: false,
})

export class ontarioIconAccessibility {

  @Prop() color: string;

  render() {
    return <div class={`ontario-icon ontario-icon--${this.color}`} innerHTML={accessibility} />;
  }
};
