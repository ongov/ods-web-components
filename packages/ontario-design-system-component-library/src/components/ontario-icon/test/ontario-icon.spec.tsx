import { newSpecPage } from '@stencil/core/testing';
import { OntarioIconAccessibility } from '../ontario-icon-accessibility';

describe('ontario-icon', () => {
  describe('snapshot', () => {
    it('should render the expected html', async () => {
      const page = await newSpecPage({
        components: [OntarioIconAccessibility],
        html: `<ontario-icon-accessibility icon-width="36" colour="grey"></ontario-icon-accessibility>`,
      });

      expect(page.root).toMatchSnapshot();
    });
  });

  it('renders', async () => {
    const page = await newSpecPage({
      components: [OntarioIconAccessibility],
      html: `<ontario-icon-accessibility icon-width="36" colour="grey"></ontario-icon-accessibility>`,
    });
    expect(page.root).toEqualHtml(`
      <ontario-icon-accessibility icon-width="36" colour="grey">
        <mock:shadow-root>
          <div class="ontario-icon ontario-icon--grey" style="width: 36px;">
          </div>
        </mock:shadow-root>
      </ontario-icon-accessibility>
    `);
  });
});
