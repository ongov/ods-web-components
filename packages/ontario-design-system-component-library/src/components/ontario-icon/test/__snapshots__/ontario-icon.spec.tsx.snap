// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ontario-icon snapshot should render the expected html 1`] = `
<ontario-icon-accessibility colour="grey" icon-width="36">
  <mock:shadow-root>
    <div class="ontario-icon ontario-icon--grey" style="width: 36px;"></div>
  </mock:shadow-root>
</ontario-icon-accessibility>
`;
